src/batchnorm.jl

```diff
@@ -0,0 +1,485 @@
+"""
+
+`BNMoments` is a high-level data structure used to store running mean and running variance
+of batch normalization.
+
+# Fields
+ `momentum::AbstractFloat`: A real number between 0 and 1 to be used as the scale of
```

Comment:
This is counterintuitive: normally high momentum means you preserve your old values more, no?

Reply:
I just checked, the original name of this parameter in cudnn is `exponentialAverageFactor`.  Is it better if we call it something like `factor`?